### PR TITLE
add nats result backend - continuation

### PR DESCRIPTION
**Pull request by rafalkrupinski**
picking up from #13 
- use official pre-commit repos for mypy and ruff
- upgraded taskiq in order to use PickleSerializer
- use standard taskiq exception
- removed old linters that were replaced by ruff

not sure if  wemake-python-styleguide is used in any way...

black could be replaced by ruff-format too, but I'll leave it for another PR, this one has enough stuff already.

### CODE REVIEW

**Comment by rafalkrupinski** [comment]
Hi, any chance anyone has a look at it? Thanks!

File: taskiq_nats/broker.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class NatsBroker(AsyncBroker):
     https://docs.nats.io/nats-concepts/core-nats/queue
     """
 
-    def __init__(  # noqa: WPS211 (too many args)
+    def __init__(  # (too many args)
```
Comment (chandr-andr) [inline taskiq_nats/broker.py]: Could you please remove `# (too many args)`? If there is no lint error anymore

File: taskiq_nats/broker.py [hunk 2]
```diff
@@ -88,7 +88,7 @@ async def shutdown(self) -> None:
         await super().shutdown()
 
 
-class BaseJetStreamBroker(  # noqa: WPS230 (too many attrs)
+class BaseJetStreamBroker(  # (too many attrs)
```
Comment (chandr-andr) [inline taskiq_nats/broker.py]:
```suggestion
class BaseJetStreamBroker(
```

File: taskiq_nats/broker.py [hunk 3]
```diff
@@ -105,7 +105,7 @@ class BaseJetStreamBroker(  # noqa: WPS230 (too many attrs)
     be sure that messages are delivered to the workers.
     """
 
-    def __init__(  # noqa: WPS211 (too many args)
+    def __init__(  # (too many args)
```
Comment (chandr-andr) [inline taskiq_nats/broker.py]:
```suggestion
    def __init__(
```

File: tests/conftest.py [hunk 4]
```diff
@@ -57,3 +61,15 @@ async def nats_jetstream(
     jetstream: Final = nats.jetstream()
     yield jetstream
     await nats.close()
+
+
+@pytest.fixture()
+async def nats_result_backend(
+    nats_urls: List[str],
+) -> AsyncGenerator[NATSObjectStoreResultBackend[_ReturnType], None]:
+    backend: NATSObjectStoreResultBackend[_ReturnType] = NATSObjectStoreResultBackend(
```
Comment (chandr-andr) [inline tests/conftest.py:70]: Is type annotation here really necessary?

**Comment by chandr-andr** [comment]
@rafalkrupinski Thank you very much for PR, in the future you can tag me if no one comes))

File: tests/conftest.py (hunk 4 above)
Comment (rafalkrupinski) [reply tests/conftest.py:70]: AFAIK fixtures are always matched by name, so not necessary from pytest point of view.

I didn't even look at this part, but I usually find the more type hints the better.

Comment (rafalkrupinski) [reply tests/conftest.py:70]: I mean for me it often helps understanding other peoples code.

Comment (chandr-andr) [reply tests/conftest.py:70]: Okay, its not a deal breaker for me)

**Review by chandr-andr — APPROVED** [review]
LGTM!

**Comment by chandr-andr** [comment]
@rafalkrupinski Thanks for the PR, I've started releasing the new version.

**Comment by chandr-andr** [comment]
@rafalkrupinski New version is here - https://pypi.org/project/taskiq-nats/0.5.0/
Thanks for the PR!

**Comment by rafalkrupinski** [comment]
@chandr-andr cheers!

**Pull request merged**